artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/ServerSessionPacketHandler.java

```diff
@@ -159,7 +158,11 @@
 
    private final boolean direct;
 
-   private static final ThreadLocal<AtomicBoolean> inHandler = ThreadLocal.withInitial(AtomicBoolean::new);
+   //marker instance used to recognize if a thread is performing a packet handling
+   private static final Object DUMMY = Boolean.TRUE;
+
+   //a thread that has its thread-local map populated with DUMMY is performing a packet handling
+   private static final ThreadLocal<Object> inHandler = new ThreadLocal<>();
```

Comment:
Because it is not needed to be a Boolean: it is just a marker, so even a DUMMY = new Object() could be used instead of Boolean to make it works.